lib/browser.js

```diff
@@ -0,0 +1,5 @@
+module.exports = typeof globalThis !== 'undefined' // new standardized access to the global object
```

Comment:
Can we use ESM for this file or CJS will be necessary? I believe that Webpack support ESM for quite a while now.

Reply:
I also think that you need to use named exports here to match library's API:

```js
import {FormData, Blob, File} from "formdata-node"
```

This should work same way both in node and browsers.
fileFromPath and fileFromPathSync can be ignored.

Reply:
CJS is bulletproof and works in all the bundlers; from what I've seen in various isomorphic libraries it's currently a standard way how to expose an isomorphic entry.

Reply:
I can see that, but ESM is the new the standard, supported by browsers natively, when CJS will eventually become obsolete. So maybe we'll convert it to ESM and see if anybody has an issue with it and then decide how to handle this problem?

Reply:
Ok let's give it try; I'll be the first tester I guess ;]

Reply:
Hm. On other hand we can only use ESM in CJS modules with `import()`. Not sure how bundlers dealt with it, but it is the way how it works in Node. Can we maybe deliver both CJS and ESM version of browser.js? As I did in the package.

Reply:
> I also think that you need to use named exports here to match library's API:
> 
> ```js
> import {FormData, Blob, File} from "formdata-node"
> ```
> 
> This should work same way both in node and browsers.
> fileFromPath and fileFromPathSync can be ignored.

Yep, you're absolutely right; I can see more symbols than `FormData, Blob, File` being exports from the library. But we need to find an intersection of what native browser env support and what this library exports; and it seems that those 3 you already mentioned are what we're looking for.

Reply:
Maybe I can just make a single TS file from browser.js and the compile it along with the whole library and add browser field to `lib/esm/package.json` and `lib/cjs/package.jsom`. Not sure if this can be resolved by bundlers that way

Reply:
We have now browser.js as ESM compatible as with what the library export (to the level of intersecting export symbols with browser env)

Reply:
Looks good to me. I'll merge it and introduce in 3.6.0 as experimental feature.

Reply:
> Maybe I can just make a single TS file from browser.js and the compile it along with the whole library and add browser field to `lib/esm/package.json` and `lib/cjs/package.jsom`. Not sure if this can be resolved by bundlers that way

Not entirely sure what you mean this this. There needs to be just one package.json for npm package AFAIK. And it needs to have one single `browser` fields to support isomorphism.

Reply:
However I'm gonna figure out it we can deliver both ESM and CJS version of this extension, just in case :)

Reply:
Just want to reiterate once more my point about `engines` in `package.json` and `ESM` for browser.js:

`engines` defined as it is forbids us to install the library on older versions of node, even though we can run it through babel after the installation

`ESM` for `browser` might disqualify using this library with older installation of webpack which runs on older installation of node.js. It's quite common problem inside big commercial organizations that we use old tools.

Given the combination of `engines` + `ESM` for `browser` field will allow to use this lib only on latest and update modern stack. The rest people that want to use this lib needs to find workarounds.

Reply:
> Maybe I can just make a single TS file from browser.js and the compile it along with the whole library and add browser field to lib/esm/package.json and lib/cjs/package.jsom. Not sure if this can be resolved by bundlers that way

I see now what you mean; needed to read: https://www.sensedeep.com/blog/posts/2021/how-to-create-single-source-npm-module.html

> However I'm gonna figure out it we can deliver both ESM and CJS version of this extension, just in case :)

Here is webpack docs about exports field: https://webpack.js.org/guides/package-exports/

`exports field is preferred over other package entry fields like main, module, browser or custom ones.`

So having `libs/browser.ts` the `exports` field of package.json would probably need to look like this:

```json
  "exports": {
    "node": {
      "import": "./lib/esm/index.js",
      "require": "./lib/cjs/index.js"
    },
    "browser": {
      "import": "./lib/esm/browser.js",
      "require": "./lib/cjs/browser.js"
    },
    "default": "./lib/esm/index.js"
  },
```

Reply:
> Not entirely sure what you mean this this. There needs to be just one package.json for npm package AFAIK. And it needs to have one single browser fields to support isomorphism.

If you look at the /lib folder, you will notice that along with the project's source code it contains two directories: `cjs` and `esm`, both of them are targets for ESM end CJS versions of the package and both of them have `package.json` file with only `type` field. I'm using [conditional exports](https://nodejs.org/dist/latest/docs/api/packages.html#packages_conditional_exports) to have this package compatible with both ESM and CJS for more smoothen transition before CJS will be gone for good in v4. I wonder if I can use this approach to deliver two versions of browser.js if I compile it with tsc, along with the /lib folder and add `browser` field to `/lib/esm` and `/lib/cjs`.

Reply:
> I see now what you mean

Yes, this looks like what I want to achieve. But for now we have browser.js :)

Reply:
> Yes, this looks like what I want to achieve. But for now we have browser.js :)

Yep trying to integrate the latest version